app/api/answers/generate_task_token.feature

```diff
@@ -0,0 +1,251 @@
+    And the groups ancestors are computed
+    And the database has the following table 'items':
+      | id | url                                                                     | type | entry_participant_type | default_language_tag | text_id |
+      | 10 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Task | User                   | fr                   | task10  |
+      | 20 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Task | Team                   | fr                   | task20  |
+      | 30 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Task | User                   | fr                   | task30  |
+      | 40 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Task | User                   | fr                   | task40  |
+      | 50 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Task | User                   | fr                   | task50  |
+    And the database has the following table 'permissions_generated':
+      | group_id | item_id | can_view_generated | can_watch_generated |
+      | 101      | 10      | content            | none                |
+      | 101      | 40      | solution           | none                |
+      | 101      | 50      | content            | none                |
+      | 102      | 20      | content            | none                |
+      | 103      | 10      | content            | answer              |
+      | 103      | 20      | content            | answer              |
+      | 103      | 30      | content            | answer              |
+      | 104      | 10      | content            | answer              |
+      | 104      | 20      | content            | answer              |
+      | 104      | 30      | content            | answer              |
+    And the database has the following table 'attempts':
+      | id | participant_id |
+      | 0  | 101            |
+      | 0  | 102            |
+      | 0  | 103            |
+      | 1  | 103            |
+    And the database has the following table 'results':
+      | attempt_id | participant_id | item_id | started_at          | validated_at        |
+      | 0          | 101            | 10      | 2020-01-01 01:01:01 | null                |
+      | 0          | 101            | 20      | 2020-01-01 01:01:01 | null                |
+      | 0          | 101            | 30      | 2020-01-01 01:01:01 | null                |
+      | 0          | 102            | 20      | 2020-01-01 01:01:01 | null                |
+      | 0          | 103            | 10      | 2020-01-01 01:01:01 | null                |
+      | 0          | 103            | 20      | 2020-01-01 01:01:01 | null                |
+      | 0          | 103            | 30      | null                | null                |
+      | 1          | 103            | 30      | 2020-01-01 01:01:01 | null                |
```

Comment:
it is a good idea to have an attempt != 0 to test that it is correct attempt_id which is used, but I don't know how you test it in this case. Maybe you could only have only one attempt "1" for this user so that we can check that it is the observed-user attempt_id which is used.

Reply:
fixed